core/src/main/java/com/linecorp/armeria/client/HttpClientPipelineConfigurator.java

```diff
@@ -268,10 +268,12 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
                     // An exception was raised before the handshake event was completed.
                     // A legacy HTTPS server such as Microsoft-IIS/8.5 may reset the connection
                     // if no cipher suites in common.
+                    final String tlsVersion = sslHandler.engine().getSession().getProtocol();
```

Comment:
Note: I checked if there's a case where `getSession()` or `getProtocol()` returns `null`, but it seems like JDK and OpenSSL implementation always return non-null.